### PR TITLE
Update log4j dependency to latest version

### PR DESCRIPTION
**Pull request by usmansaleem**

### CODE REVIEW

**Review by jframe — APPROVED** [review]
LGTM

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,5 +1,9 @@
 #Changelog
 
+## 1.0.22-24
+### Bug Fixed
+- Updated log4j library versions
```
Comment (siladu) [inline CHANGELOG.md:5]: Might be useful to specify the version we upgraded to

Comment (usmansaleem) [reply CHANGELOG.md]: done

File: CHANGELOG.md [hunk 2]
```diff
@@ -1,5 +1,11 @@
 #Changelog
 
+## 1.0.24
+### Bugs Fixed
+- Updated to log4j 2.17.1. Resolves two potential vulnerabilities which are only exploitable when using custom log4j configurations that are either writable by untrusted users or log data from the `ThreadContext`.
+
+---
```
Comment (jframe) [inline CHANGELOG.md:7]: What happened to change logs for between 1.0.21 and 1.0.24?

Comment (usmansaleem) [reply CHANGELOG.md:7]: they weren't created ... and all of them updated log4j versions update as well

Comment (usmansaleem) [reply CHANGELOG.md:7]: let me add them.

**Review by siladu — APPROVED** [review]
LGTM

**Pull request merged**